### DISCUSSION TITLE
Can't type some symbols

### DISCUSSION THREAD

**Discussion by lukewilde** [manna-harbour/miryoku#197] [Q&A] [answered]
Hey!

When I press the following keys on the symbol layer I get the following output:

![image](https://user-images.githubusercontent.com/856208/223471639-d0cb3fd9-cf0d-484e-87d3-9820645177df.png)

1. `~`
2. `¬`
3. `£`
4. `"` (this one differs from the layout graphic but it's in the correct position for UK keyboards)

And when I press the first key on the num layer, it outputs `#`.

I'm using the ZMK version of the layout on Windows 11. 

I've checked the ZMK, Miryoku and Miryoku-zmk docs for anything relating to language / region but can't find any relevant config.

I've also double-checked with my old keyboard and it can output the keys correctly, so I don't think it's an OS level config I've messed up.

Thanks in advance!

**Comment by ntzm** [comment]
I have the same issue with qmk on Linux!

**Comment by manna-harbour** [comment, marked as answer]
Keyboard firmware assumes the host is using US ANSI. For localised layouts on the host you need to translate keycodes.

For QMK see https://docs.qmk.fm/#/reference_keymap_extras?id=header-files. 
For ZMK see https://github.com/joelspadin/zmk-locale-generator/releases. 

For Miryoku, you'd add a `#include` for the appropriate localisation header file to `custom_config.h` and then substitute custom layers with the localised keycodes where they differ. See https://github.com/manna-harbour/miryoku/discussions/85.

Alternatively, use EurKEY, Compose, or US International, which wouldn't require any modification to the keymap.

**Reply by ntzm** [reply]
Thanks, for some reason I get compile errors when I include keymap_uk in my custom config, however this works well for me (for anyone who wants it in the future):

in custom_config.h

```c
// layout
// used in .c file to make shift quote work
#define UK_DQUO S(KC_2) // "

// used below to make symbols
#define UK_BSLS KC_NUBS //
#define UK_HASH KC_NUHS    // #
#define UK_AT   S(KC_QUOT) // @
#define UK_TILD S(UK_HASH) // ~
#define UK_PIPE S(UK_BSLS) // |

#define MIRYOKU_LAYER_SYM \
KC_LCBR,           KC_AMPR,           KC_ASTR,           KC_LPRN,           KC_RCBR,           U_NA,              TD(U_TD_U_BASE),   TD(U_TD_U_EXTRA),  TD(U_TD_U_TAP),    TD(U_TD_BOOT),     \
KC_COLN,           KC_DLR,            KC_PERC,           KC_CIRC,           KC_PLUS,           U_NA,              KC_LSFT,           KC_LCTL,           KC_LALT,           KC_LGUI,           \
UK_TILD,           KC_EXLM,           UK_AT,             UK_HASH,           UK_PIPE,           U_NA,              TD(U_TD_U_SYM),    TD(U_TD_U_MOUSE),  KC_ALGR,           U_NA,              \
U_NP,              U_NP,              KC_LPRN,           KC_RPRN,           KC_UNDS,           U_NA,              U_NA,              U_NA,              U_NP,              U_NP

#define MIRYOKU_LAYER_NUM \
KC_LBRC,           KC_7,              KC_8,              KC_9,              KC_RBRC,           U_NA,              TD(U_TD_U_BASE),   TD(U_TD_U_EXTRA),  TD(U_TD_U_TAP),    TD(U_TD_BOOT),     \
KC_SCLN,           KC_4,              KC_5,              KC_6,              KC_EQL,            U_NA,              KC_LSFT,           KC_LCTL,           KC_LALT,           KC_LGUI,           \
KC_GRV,            KC_1,              KC_2,              KC_3,              UK_BSLS,           U_NA,              TD(U_TD_U_NUM),    TD(U_TD_U_NAV),    KC_ALGR,           U_NA,              \
U_NP,              U_NP,              KC_DOT,            KC_0,              KC_MINS,           U_NA,              U_NA,              U_NA,              U_NP,              U_NP
```

then to make it so that shift + ' makes " instead of @, in manna-harbour_miryoku.c

```diff
--- a/users/manna-harbour_miryoku/manna-harbour_miryoku.c
+++ b/users/manna-harbour_miryoku/manna-harbour_miryoku.c
@@ -53,8 +53,11 @@ MIRYOKU_LAYER_LIST
 
 const key_override_t capsword_key_override = ko_make_basic(MOD_MASK_SHIFT, CW_TOGG, KC_CAPS);
 
+const key_override_t double_quote_key_override = ko_make_basic(MOD_MASK_SHIFT, KC_QUOT, UK_DQUO);
+
 const key_override_t **key_overrides = (const key_override_t *[]){
     &capsword_key_override,
+    &double_quote_key_override,
     NULL
 };
```
